language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaDocumentServiceImpl.java

```diff
@@ -337,7 +341,17 @@ public BallerinaDocumentServiceImpl(WorkspaceManager workspaceManager, LanguageS
                 }
                 SyntaxTree syntaxTree = this.workspaceManager.syntaxTree(filePath.get()).orElseThrow();
                 NonTerminalNode currentNode = CommonUtil.findNode(params.getRange(), syntaxTree);
-                syntaxTreeNodeResponse.setKind(currentNode.kind().name());
+                LinePosition startLine = currentNode.lineRange().startLine();
```

Comment:
Shall we add some test cases to cover the faulty cases as well?

Reply:
Updated.